### PR TITLE
Avatars with autoicons

### PR DESCRIPTION
**Pull request by iafan**
This PR introduces:
- AutoIcon component to render nice avatar-style 'icons' based on provided string
- Improved Avatar component that makes use of AutoIcon
- Navbar menu icon is now rendered using Avatar component

Note that there are several more areas that require moving to Avatar component for consistency: welcome page (top contributors list) and editor.

### CODE REVIEW

File: pootle/static/js/shared/components/AutoIcon.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+ * This file is a part of the Zing project. It is distributed under the GPL3
+ * or later license. See the LICENSE file for a copy of the license and the
+ * AUTHORS file for copyright and authorship information.
+ */
+
+import React from 'react';
+
+import crc16 from '../utils/crc16'; // use relative path, otherwise tests will fail
+
+export function calcAbbreviation(title, length) {
```
Comment (julen) [inline pootle/static/js/shared/components/AutoIcon.js]: I understand the reason to export this function (and the other one below) is for tests, although it is not necessary at all as we use the `rewire` plugin in tests, so they can retrieve private module dependencies. See https://github.com/evernote/zing/commit/da3ba2a for an example on how to use it.

File: pootle/static/js/shared/components/AutoIcon.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      abbreviationLength: React.PropTypes.number,
+      mode: React.PropTypes.string,
+      size: React.PropTypes.number,
+    };
+  }
+
+  constructor(props) {
+    super();
+
+    this.state = Object.assign({
```
Comment (julen) [inline pootle/static/js/shared/components/AutoIcon.js]: Using `Object.assign` directly will definitely break IE11. Unless we plan to stop supporting it in the near future, please use `assign` from the `object-assign` package instead (`import assign from 'object-assign';`).

File: pootle/static/js/shared/components/Avatar.js [hunk 3]
```diff
@@ -6,71 +6,91 @@
...
+          </span>
+          <span className="user-name" title={username}>{title}</span>
+        </TagName>
       );
     }
 
-    return icon;
+    return (
+      <TagName className="avatar">
+          <span className="avatar-image" style={style}>
```
Comment (julen) [inline pootle/static/js/shared/components/Avatar.js]: I'm not sure how the linter didn't kick in, but we need to use 2-space indents consistently.

File: pootle/static/js/shared/components/AutoIcon.js (hunk 2 above)
Comment (iafan) [inline pootle/static/js/shared/components/AutoIcon.js]: This code was already changed to `_.extend()` in bb8de4971544b6374f4cb98e02e73a896a65d1a0

I guess `_.extend()` is more appropriate in this case of extending properties?

File: pootle/static/js/shared/components/AutoIcon.js (hunk 1 above)
Comment (iafan) [inline pootle/static/js/shared/components/AutoIcon.js]: Fixed in 5fbd2ea8a3e78577eb5d39ba572abbd7ffa5023d

File: pootle/static/js/shared/components/AutoIcon.js (hunk 2 above)
Comment (julen) [inline pootle/static/js/shared/components/AutoIcon.js]: We have been using `assign` from the `object-assign` package as it's a drop-in replacement of the `Object.assign()` functionality and we should favor it over other variations. Occurrences of `_.extend()` as well as `$.extend()`can be considered old code.

Comment (iafan) [inline pootle/static/js/shared/components/AutoIcon.js]: Changed in cd79bdf8673315edcb1d496e82d9bc7f4e4811c5

**Pull request merged**